### DISCUSSION TITLE
Better no-explicit-any - support JSDoc @type

### DISCUSSION THREAD

**Discussion by alebas1** [biomejs/biome#1594] [Ideas]
Hello, the use of JSDoc instead of typescript is growing in the JS ecosystem. So I tried to configure biome on a project that uses only JSDoc and no TS.

I came across an issue using biome's linter.

When defining something like:
```js
let foo;
```
I should be getting an error from the [no-explicit-any](https://biomejs.dev/linter/rules/no-explicit-any) rule, since foo is of type any.

To fix that i would need to defined a type using JSDoc:
```js
/** @type {string} */
let foo;
```

I know changing this [no-explicit-any](https://biomejs.dev/linter/rules/no-explicit-any) rule would probably be a breaking change, but to me, it would be a great feature (that could be implemented by a new rules specific to javascript or by a rule option?)

**Comment by nhedger** [comment]
> When defining something like:
> 
> ```js
> let foo;
> ```
> I should be getting an error from the [no-explicit-any](https://biomejs.dev/linter/rules/no-explicit-any) rule, since foo is of type any.

Your snippet being JavaScript, the concept of `any` does not apply. Even if this was TypeScript, the `no-explicit-any` rule would not trigger because `any` would be _implicit_ in this case.

https://biomejs.dev/playground/?code=bABlAHQAIABmAG8AbwA7AA%3D%3D&jsx=false
```
main.ts:1:5 [lint/suspicious/noImplicitAnyLet](https://biomejs.dev/linter/rules/no-implicit-any-let) ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This variable has implicitly the any type.
  
  > 1 │ let foo;
      │     ^^^
  
  ℹ Variable declarations without type annotation and initialization have implicitly the any type. Declare type or initialize the variable with some value.
```

Am I missing something?
